### DISCUSSION TITLE
Using useDocument in a pinia store

### DISCUSSION THREAD

**Discussion by starcraft04** [vuejs/vuefire#1422] [Questions and Help] [answered]
I am trying to use the useDocument from VueFire in a pinia store in composition API and I have tried different ways but none are working.
I want to make sure I initiate it in the function and not at the start of the pinia file... This is because I need to get the data from firebase only when the function getAllProfiles is called.

I am using the following:
Pinia Store
____________

export const useAllProfileStore = defineStore('allProfiles', () => {
  const oneDoc = ref({})
  const oneDocLoading = ref(false)

async function getAllProfiles() {
    const { pending } = useDocument(doc(db, 'members', 'xxx'), { target: oneDoc })
  }

return {
    oneDoc,
    oneDocLoading,
...

Main page
___________
/* imports */
import { onMounted,watch } from 'vue'
import { useAllProfileStore } from '@/stores/allProfiles'
import { storeToRefs } from 'pinia'

/* Const */
const allProfileStore = useAllProfileStore()
const { oneDoc, oneDocLoading } = storeToRefs(allProfileStore)

watch(()=>oneDoc,()=>{console.log(oneDoc.value);
},{deep:true})

async function initTable() {
  await allProfileStore.getAllProfiles()
}

Just trying to get the data, having it in a realtime and be able to share it accross different components. 

!!!!My problem is that I will need to assign the pending state to the pinia ref oneDocLoading.!!!!

I really tried so many different variations but none are working...

**Comment by posva** [comment, marked as answer]
Like other composables [you must call them within injectable contexts](https://vuefire.vuejs.org/guide/other-firebase-services.html#Firebase-Services). In the case of pinia stores, this means at the root level of the function.

**Reply by starcraft04** [reply]
Ok, so you mean I have no choice but to call it in the init phase of the pinia store with:
export const useAllProfileStore = defineStore('allProfiles', () => {
  const { data: oneDoc, pending: oneDocloading } = useDocument(doc(db, 'members', 'xxx'))

I have tried this, it works but then it means that I have no choice to issue the network call and load the data from the db at the init of the store. I cannot get this call with vueFire later on for example when the user has clicked a button in one of the components. 

IS it correct that the only way to use useDocument from vuefire is at the start of the setup of the pinia and not in a function from a pinia store...

**Reply by posva** [reply]
Note the first argument of useDocument can be a ref and that it can be null too. You could change the value of the ref on click
